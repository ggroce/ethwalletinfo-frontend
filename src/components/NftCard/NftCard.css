.nftcard__wrapper {
  border: 1px solid;
  border-radius: 10px;
  border-color: rgb(117, 110, 110);
  background-color: #191b20;
  width: 12rem;
  height: 17rem;
  line-height: 1.5;
  margin: 1rem;
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  font-size: medium;
  cursor: pointer;
  box-sizing: border-box;
  overflow: hidden;
}

.nftcard__info-wrapper {
  padding: 10px;
  max-height: 3rem;
  text-align: center;
  overflow: hidden;
}

.nftcard__img {
  display: block;
  margin-left: auto;
  margin-right: auto;
  max-width: 10rem;
  max-height: 10rem;
  width: auto;
  height: auto;
}
