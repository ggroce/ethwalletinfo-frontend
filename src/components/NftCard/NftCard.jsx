import './NftCard.css';
import NftDetailsModal from '../NftDetailsModal/NftDetailsModal';
import { useState } from 'react';

const NftCard = (props) => {
  const [showDetails, setShowDetails] = useState(false);

  return (
    <>
      <div className="nftcard__wrapper" onClick={() => setShowDetails(true)}>
        <div className="nftcard__info-wrapper">
          <h5>{props.meta?.name}</h5>
        </div>
        <img
          className="nftcard__img"
          src={props.meta?.image.url.PREVIEW || props.meta?.image.url.ORIGINAL}
          loading="lazy"
          title="Picture of NFT"
          alt={props.meta?.name}
        />
      </div>
      {showDetails && (
        <NftDetailsModal setShowDetails={setShowDetails} props={props} />
      )}
    </>
  );
};
export default NftCard;
