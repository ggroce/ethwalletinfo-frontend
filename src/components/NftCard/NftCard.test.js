import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NftCard from './NftCard';

const mockNft = {
  meta: {
    name: 'Random NFT',
    description: "This NFT is the bee's knees",
    image: {
      url: {
        PREVIEW: 'https://example.com/preview.png',
        ORIGINAL: 'https://example.com/original.png',
      },
    },
  },
};

test('<NftCard />: takes a snapshot', () => {
  const { asFragment } = render(<NftCard />);
  expect(asFragment(<NftCard />)).toMatchSnapshot();
});

test('<NftCard />: checks for correct name on NFT title', () => {
  render(<NftCard {...mockNft} />);
  expect(screen.getByRole('heading')).toHaveTextContent('Random NFT');
});

test('<NftCard />: checks for correct modal information and open/close on click', async () => {
  render(<NftCard {...mockNft} />);
  userEvent.click(screen.getByRole('heading'));
  await waitFor(() =>
    expect(
      screen.getByText("Description: This NFT is the bee's knees")
    ).toBeInTheDocument()
  );
  userEvent.click(screen.getByText("Description: This NFT is the bee's knees"));
  await waitFor(() =>
    expect(
      screen.queryByText("Description: This NFT is the bee's knees")
    ).not.toBeInTheDocument()
  );
});
