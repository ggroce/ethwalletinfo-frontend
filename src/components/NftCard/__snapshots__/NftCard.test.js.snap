// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NftCard />: takes a snapshot 1`] = `
<DocumentFragment>
  <div
    class="nftcard__wrapper"
  >
    <div
      class="nftcard__info-wrapper"
    >
      <h5 />
    </div>
    <img
      class="nftcard__img"
      loading="lazy"
      title="Picture of NFT"
    />
  </div>
</DocumentFragment>
`;
